Source/JavaScriptCore/assembler/MacroAssemblerARMv7.h

```diff
@@ -191,6 +191,49 @@ class MacroAssemblerARMv7 : public AbstractMacroAssembler<Assembler> {
     static constexpr RegisterID framePointerRegister = ARMRegisters::fp;
     static constexpr RegisterID linkRegister = ARMRegisters::lr;
 
+private:
+    template<typename instruction>
+    void execConcurrently(instruction insn)
+    {
+        // We need 2 temp registers to make the lock/unlock work
+        // We don't call getAndInvalidate here because after the lock/unlock we can reuse
+        // them in other calls, e.g., storePair32, branch32, setupArmAddress, etc.
+        constexpr RegisterID scratch = dataTempRegister;
+        constexpr RegisterID scratch2 = addressTempRegister;
+
+        ArmAddress lockAddress = setupArmAddress(AbsoluteAddress(&globalLock), scratch2);
+
+        Label tryLock = label();
+        m_assembler.ldrex(scratch, lockAddress.base, ARMThumbImmediate::makeUInt12(lockAddress.u.offset));
+        Jump lockAcquired = branch32(NotEqual, scratch2, TrustedImm32(0));
+
+        // Here we are actually writting the address of the lock at the address of the lock
+        // While gcc/clang write #1 to indicate that we have the lock, we don't really care about the value
+        // written, since we only need it to be not #0 for the ldrex to work
+        m_assembler.strex(scratch, scratch2, lockAddress.base, ARMThumbImmediate::makeUInt12(lockAddress.u.offset));
+        branch32(NotEqual, scratch, TrustedImm32(0)).linkTo(tryLock, this);
```

Comment:
Neat, but I think if lockAddress.base happens to be scratch due to the cached register value fitting the offset, then that becomes illegal, so please make sure that does not happen.